sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -110,17 +110,34 @@ export interface EventDataInternal {
   /**
    * @property [systemProperties] The properties set by the service.
    */
-  systemProperties?: Dictionary<any>;
+  systemProperties: { [property: string]: any };
 }
 
+const messagePropertiesMap = {
+  message_id: "messageId",
+  user_id: "userId",
+  to: "to",
+  subject: "subject",
+  reply_to: "replyTo",
+  correlation_id: "correlationId",
+  content_type: "contentType",
+  content_encoding: "contentEncoding",
+  absolute_expiry_time: "absoluteExpiryTime",
+  creation_time: "creationTime",
+  group_id: "groupId",
+  group_sequence: "groupSequence",
+  reply_to_group_id: "replyToGroupId"
+} as const;
+
 /**
  * Converts the AMQP message to an EventData.
  * @param msg The AMQP message that needs to be converted to EventData.
  * @ignore
  */
 export function fromAmqpMessage(msg: Message): EventDataInternal {
   const data: EventDataInternal = {
-    body: msg.body
+    body: msg.body,
+    systemProperties: {}
```

Comment:
Any concern that in the absence of any system properties, this would be `undefined` before this PR and moving forward, it would be empty object?

Reply:
I don’t think so. Really, I think it will nearly, if not always be defined now given all the new properties that can be added (for example, the `to` field). This reduces the number of times we had to check if the object already existed and doesn't hurt the API (if anything I think of we made it a required property that makes things easier by removing an existence check.)

That said, it could lead to more objects being allocated which isn’t free if none of the properties exist. I was testing with iothub events but can check with plain event hub events to see if they are ever all undefined.

Reply:
>  I was testing with iothub events but can check with plain event hub events to see if they are ever all undefined

Lets do that

Reply:
All the new properties can be undefined if sending a message with just a body using to event hubs.

I still think it's ok to return an empty object for systemProperties but I can revert that if you have concerns.

Reply:
Reverted. Favoring memory optimization over (potentially) compute.